### ISSUE TITLE
Documentation: types vs codecs confusing

### ISSUE THREAD

**Issue by osdiab** [gcanti/io-ts#331]
**Do you want to request a _feature_ or report a _bug_?**
More of a suggestion, not really either

**What is the current behavior?**
README and names of types call `io-ts` codecs "types" which is how they behave, but I find it gets confusing when you're talking about io-ts types (which are not typescript types) and typescript types (which are not io-ts types) and none of the actual code uses the word "codecs" which would have been much clearer.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://codesandbox.io/ or similar.**
N/A

**What is the expected behavior?**
Having clearly separate nomenclature than "types", i.e. rename `Type<A, O, I>` to `Codec<A, O, I>`, maybe rename `t.type()` to `t.shape()` or something, and reserve the word `Type` for only referring to TypeScript types (so `TypeOf<Codec>` is fine).

then only refer to `io-ts` codecs and there's no nomenclature confusion.

Though doing this would probably break a bunch of stuff so I mostly just want to bring attention to the issue than actually make the change.

**Which versions of io-ts, and which browser and OS are affected by this issue? Did this work in previous versions of io-ts?**
All of them

**Comment by gcanti** [comment]
> Though doing this would probably break a bunch of stuff so I mostly just want to bring attention to the issue than actually make the change

@osdiab yeah, I'm waiting to start working on v2 for this kind of changes

> rename Type<A, O, I> to Codec<A, O, I>

:+1:

> maybe rename t.type() to t.shape()

:-1:

the name `type` refers to type aliases

```ts
//               v--- this is the runtime equivalent of...
const Person = t.type({
  name: t.string
})

// v--- ...this
type Person = {
  name: string
}
```

**Comment by osdiab** [comment]
Yeah I was thinking that typescripts type alias keyword is very very flexible

type x = number
type y = ReturnType<some function>

Etc. if anything interface may be more akin to the meaning. But yeah it’d be pretty breaking. Maybe just offering alternate syntax that allows one to avoid the overloaded name haha
